rpcs3/Emu/Cell/Modules/cellSsl.cpp

```diff
@@ -52,11 +60,15 @@ s32 cellSslCertificateLoader(u64 flag, vm::ptr<char> buffer, u32 size, vm::ptr<u
 	if (required)
 	{
 		*required = (u32)flagBits.count();
```

Comment:
Btw, does this implementation account for a `null` character? I.e. does it return the required length for a buffer + a `\0` char? If not, this might be the reason for TLoU crash.

Reply:
Required does pass the correct value, I wrote a test for it here: 
https://github.com/hcorion/ps3-homebrew/tree/master/ssl_certificate_test-tlou

Reply:
@hcorion is `End of array is` there the same as `size: `?

Reply:
No. From what I remember size is `End of array` + the number of certs not loaded.

Reply:
That's strange, since fromhttps://github.com/hcorion/ps3-homebrew/blob/master/ssl_certificate_test-tlou/source/main.c#L112 `size` should be exactly the required length of the buffer to store certs, so here https://github.com/hcorion/ps3-homebrew/blob/master/ssl_certificate_test-tlou/source/main.c#L43 it should print the index past the cert loaded to `buf` - i.e. `size`.

Reply:
Sure. In theory it should do that, but from my findings, as I said, it doesn't seem to be what HW/LLE Ssl does.

If you can write some tests to prove my theory wrong, go ahead, I'd be happy to be proven wrong.

Reply:
@hcorion I don't have ps3 HW available, so I'm not sure how I would be able to write proper tests) Btw, it looks like https://github.com/hcorion/ps3-homebrew/blob/master/ssl_certificate_test/console-output.txt is outdated (i.e. shouldn't the size be printed for every cert with the current code?)

Reply:
Just test against LLE Ssl. 
Yeah, console-output.txt is outdated.